### PR TITLE
Log api calls

### PR DESCRIPTION
**Pull request by StupidCodeFactory**
https://trello.com/c/L2s8TdDi/737-5-log-api-calls-made-from-pvb2

**CHANGE**:

I've imported the ```Instrumentation``` class from pvb-public.

### CODE REVIEW

File: spec/services/nomis/client_spec.rb [hunk 1]
```diff
@@ -15,19 +15,44 @@
...
+        to_raise(Excon::Errors::Timeout.new('Request Timeout'))
+    end
+
+    it 'raises an APIError if an unexpected exception is raised containing request information' do
+      expect {
+        subject.get(path, params)
+      }.to raise_error(Nomis::APIError, 'Exception Excon::Errors::Timeout calling GET /nomisapi/lookup/active_offender: Request Timeout')
+    end
+
+    it 'imcrements the api error count' do
```
Comment (alan) [inline spec/services/nomis/client_spec.rb]: increments

**Comment by alan** [comment]
Can you squash the commits as there is a lot of wip commits?

**Comment by StupidCodeFactory** [comment]
Yes there are few challenges with API's api, it's rather difficult to hook into it. 
Also i've had a look at the ```excon``` gem it does not do connection pooling and keep alive out of the box. Implementing this could potentially reduce the amount of connection errors we have.

**Comment by alan** [comment]
We are using persistent connections with exconn, which I think is what you mean by keep alive?Agree with the connection pool, this is something we have with the sendgrid api client, we could do something similar.  Without it we'll potentially have contention between puma threads when making API calls.

**Pull request merged**